test/functional/wallet_assumeutxo.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+# Copyright (c) 2023-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test for assumeutxo wallet related behavior.
+See feature_assumeutxo.py for background.
+
+## Possible test improvements
+
+- TODO: test import descriptors while background sync is in progress
+- TODO: test loading a wallet (backup) on a pruned node
+
+"""
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    assert_raises_rpc_error,
+)
+
+START_HEIGHT = 199
+SNAPSHOT_BASE_HEIGHT = 299
+FINAL_HEIGHT = 399
+
+
+class AssumeutxoTest(BitcoinTestFramework):
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def add_options(self, parser):
+        self.add_wallet_options(parser, legacy=False)
+
+    def set_test_params(self):
+        """Use the pregenerated, deterministic chain up to height 199."""
+        self.num_nodes = 2
+        self.rpc_timeout = 120
+        self.extra_args = [
+            [],
+            [],
+        ]
+
+    def setup_network(self):
+        """Start with the nodes disconnected so that one can generate a snapshot
+        including blocks the other hasn't yet seen."""
+        self.add_nodes(2)
+        self.start_nodes(extra_args=self.extra_args)
+
+    def run_test(self):
+        """
+        Bring up two (disconnected) nodes, mine some new blocks on the first,
+        and generate a UTXO snapshot.
+
+        Load the snapshot into the second, ensure it syncs to tip and completes
+        background validation when connected to the first.
+        """
+        n0 = self.nodes[0]
+        n1 = self.nodes[1]
+
+        # Mock time for a deterministic chain
+        for n in self.nodes:
+            n.setmocktime(n.getblockheader(n.getbestblockhash())['time'])
+
+        self.sync_blocks()
+
+        n0.createwallet('w')
+        w = n0.get_wallet_rpc("w")
+
+        # Generate a series of blocks that `n0` will have in the snapshot,
+        # but that n1 doesn't yet see. In order for the snapshot to activate,
+        # though, we have to ferry over the new headers to n1 so that it
+        # isn't waiting forever to see the header of the snapshot's base block
+        # while disconnected from n0.
+        for _ in range(100):
+            self.generate(n0, nblocks=1, sync_fun=self.no_op)
+            newblock = n0.getblock(n0.getbestblockhash(), 0)
+
+            # make n1 aware of the new header, but don't give it the block.
+            n1.submitheader(newblock)
+
+        # Ensure everyone is seeing the same headers.
+        for n in self.nodes:
+            assert_equal(n.getblockchaininfo()[
+                         "headers"], SNAPSHOT_BASE_HEIGHT)
+
+        w.backupwallet("backup_w.dat")
```

Comment:
From reading the test again (it's a been a while) I don't think we care about this loaded wallet `w` anymore.

Reply:
I don't think the problem is that the original wallet is open. The error is the path for the restored wallet on node1, not the original on node0.

Reply:
I see (just saw the log in #29234).

Sounds like the node that made the backup holds on to it after writing? Maybe restart the node? Or copy the backup file?

Reply:
> Sounds like the node that made the backup holds on to it after writing? Maybe restart the node? Or copy the backup file?

We don't remove the backup. The failure is in cleaning up the restored wallet after it fails to restore. The backup file is copied to a new path, and that is what `remove_all` is trying to remove.

I think it's more likely that the database isn't fully closed yet even though the `CWallet` should be destructed.

Reply:
It looks like we don't have any tests that have a failure in `CWallet::Create`, which I don't think is that surprising since generally the wallet needs to be corrupted for that to happen.

Reply:
> I think it's more likely that the database isn't fully closed yet even though the `CWallet` should be destructed.

Maybe it's because of `-unsafesqlitesync`, going to try turning that off for the test.

Reply:
I wonder why it's called unsafe :-)